packages/dapp-svelte-wallet/api/src/wallet.js

```diff
@@ -49,167 +50,150 @@ export function buildRootObject(_vatPowers) {
     pushOfferSubscriptions(channelHandle, inboxState);
   };
 
-  const {
-    updater: pursesJSONUpdater,
-    notifier: pursesJSONNotifier,
-  } = makeNotifierKit(pursesState);
-  const { publish: pursesPublish, subscribe: purseSubscribe } = pubsub(E);
-  const {
-    updater: inboxJSONUpdater,
-    notifier: inboxJSONNotifier,
-  } = makeNotifierKit(inboxState);
+  const { publish: pursesPublish, subscribe: pursesSubscribe } = pubsub(E);
+  const { updater: inboxUpdater, notifier: inboxNotifier } = makeNotifierKit(
+    inboxState,
+  );
   const { publish: inboxPublish, subscribe: inboxSubscribe } = pubsub(E);
 
   const notifiers = harden({
-    getInboxJSONNotifier() {
-      return inboxJSONNotifier;
-    },
-    getPursesJSONNotifier() {
-      return pursesJSONNotifier;
+    getInboxNotifier() {
+      return inboxNotifier;
     },
   });
 
   async function startup({ zoe, board }) {
-    wallet = await makeWallet({
+    walletInternals = await makeWallet({
       zoe,
       board,
       pursesStateChangeHandler: pursesPublish,
       inboxStateChangeHandler: inboxPublish,
     });
   }
 
-  async function getWallet() {
-    return harden({ ...wallet, ...notifiers });
-  }
-
-  function setHTTPObject(o, _ROLES) {
-    http = o;
-  }
-
-  async function adminOnMessage(obj, meta = { origin: 'unknown' }) {
-    const { type, data, dappOrigin = meta.origin } = obj;
-    switch (type) {
-      case 'walletGetPurses': {
-        return {
-          type: 'walletUpdatePurses',
-          data: pursesState || {},
-        };
-      }
-      case 'walletGetInbox': {
-        return {
-          type: 'walletUpdateInbox',
-          data: inboxState || {},
-        };
+  const makeBridge = (approve, dappOrigin, meta = {}) => {
+    async function* makeApprovedNotifier(sourceNotifier) {
+      await approve();
+      for await (const state of sourceNotifier) {
+        await approve();
+        yield state;
       }
-      case 'walletAddOffer': {
-        let handled = false;
-        const actions = harden({
-          result(offer, outcome) {
-            httpSend(
-              {
-                type: 'walletOfferResult',
-                data: {
-                  id: offer.id,
-                  dappContext: offer.dappContext,
-                  outcome,
-                },
-              },
-              [meta.channelHandle],
-            );
+    }
+
+    /** @type {WalletBridge} */
```

Comment:
I think we need comments on what the facets are and where they are expected to be used.

Reply:
By bridge, I assume this is what is available to approved dapps? Let's make sure to explain that.

Reply:
> By bridge, I assume this is what is available to approved dapps? Let's make sure to explain that.

Will do.